23.06 Install Page Update (#396)
rapidsai/docs: install/install.md

<br/>

### **pip Issues**

> <i class="fas fa-info-circle"></i> The CUDA version on your system must match the pip CUDA version you install (`-cu11` or `-cu12`). If you're unsure of your system CUDA version, the code below automatically determines it:<br/>

Reviewer: Can we verify this? This will install packages that match the driver and not the toolkit, if I understand the behavior of `nvidia-smi` correctly. I believe if you install CUDA 12 drivers (e.g. 525) and a CUDA 11 toolkit, this will install an incompatible version of RAPIDS (it'll install `-cu12` but you need to match the toolkit, `-cu11`). You can test this on some of the ASE lab machines, which have driver 525 and CUDA 11.8 toolkit installed.

Reviewer: Yeah... I wouldn't rely on CUDA being installed to `/usr/local`. Can we just say "install the version that matches your CUDA toolkit version" and not attempt to write user-proof code to do it automatically?

Reviewer: (For a counter-case of the `/usr/local` code, you could install the CUDA toolkit via conda and not install a system CTK, only a driver.)

Author: Yup, agree we should just leave it to the user to match them. If we get lots of issues filed due to challenges here we can reconsider.

<br/>

### **pip Issues**

> <i class="fas fa-info-circle"></i> pip installations require using the matching wheel to the system’s installed CUDA toolkit. For CUDA 11 toolkits, install the `-cu11` wheels, and for CUDA 12 tookits install the `-cu12` wheels. If your installation has a CUDA 12 driver but a CUDA 11 toolkit, use the `-cu11` wheels. <br/>

Reviewer: Avoid smart (curly) quotes.
```suggestion
<i class="fas fa-info-circle"></i> pip installations require using the matching wheel to the system's installed CUDA toolkit. For CUDA 11 toolkits, install the `-cu11` wheels, and for CUDA 12 tookits install the `-cu12` wheels. If your installation has a CUDA 12 driver but a CUDA 11 toolkit, use the `-cu11` wheels. <br/>
```

- <i class="fas fa-check-circle"></i> [CUDA 11.4](https://developer.nvidia.com/cuda-11-4-0-download-archive){: target="_blank"} with Driver 470.42.01 or newer
- <i class="fas fa-check-circle"></i> [CUDA 11.5](https://developer.nvidia.com/cuda-11-5-0-download-archive){: target="_blank"} with Driver 495.29.05 or newer
- <i class="fas fa-check-circle"></i> [CUDA 11.8](https://developer.nvidia.com/cuda-11-8-0-download-archive){: target="_blank"} with Driver 520.61.05 or newer
- <i class="fas fa-check-circle"></i> [CUDA 12.0](https://developer.nvidia.com/cuda-12-0-1-download-archive){: target="_blank"} with Driver 525.60.13 or newer **for pip installations only**

 **Note**: RAPIDS is tested with and officially supports the versions listed above. Newer CUDA and driver versions may also work with RAPIDS. See [CUDA compatibility](https://docs.nvidia.com/deploy/cuda-compatibility/index.html) for details. 
 

>  **Note**: RAPIDS Docker and Conda CUDA 11 environments function as expected on a system with a CUDA 12 driver and they include their own toolkit. pip installations require using the matching wheel to the system's installed CUDA toolkit. For CUDA 11 toolkits, install the <code>-cu11</code> wheels, and for CUDA 12 tookits install the <code>-cu12</code> wheels. If your installation has a CUDA 12 driver but a CUDA 11 toolkit, use the <code>-cu11</code> wheels.

Reviewer: ```suggestion
 **Note**: RAPIDS Docker images and conda packages for CUDA 11 can be used on a system with a CUDA 12 driver because they include their own CUDA toolkit. pip installations require using a wheel matching the system's installed CUDA toolkit. For CUDA 11 toolkits, install the <code>-cu11</code> wheels, and for CUDA 12 toolkits install the <code>-cu12</code> wheels. If your installation has a CUDA 12 driver but a CUDA 11 toolkit, use the <code>-cu11</code> wheels.
```

- <i class="fas fa-check-circle"></i> [CUDA 11.4](https://developer.nvidia.com/cuda-11-4-0-download-archive){: target="_blank"} with Driver 470.42.01 or newer
- <i class="fas fa-check-circle"></i> [CUDA 11.5](https://developer.nvidia.com/cuda-11-5-0-download-archive){: target="_blank"} with Driver 495.29.05 or newer
- <i class="fas fa-check-circle"></i> [CUDA 11.8](https://developer.nvidia.com/cuda-11-8-0-download-archive){: target="_blank"} with Driver 520.61.05 or newer

> - <i class="fas fa-check-circle"></i> [CUDA 12.0](https://developer.nvidia.com/cuda-11-8-0-download-archive){: target="_blank"} with Driver 525.60.13 or newer **for pip installations only**

Reviewer: Fix URL.
```suggestion
- <i class="fas fa-check-circle"></i> [CUDA 12.0](https://developer.nvidia.com/cuda-12-0-1-download-archive){: target="_blank"} with Driver 525.60.13 or newer **for pip installations only**
```

 **Note**: RAPIDS is tested with and officially supports the versions listed above. Newer CUDA and driver versions may also work with RAPIDS. See [CUDA compatibility](https://docs.nvidia.com/deploy/cuda-compatibility/index.html) for details. 
 

>  **Note**: Docker and Conda CUDA 11.x installations function as expected on a CUDA 12.x machine. pip installations require using the matching wheel to the system's installed CUDA. For CUDA 11.x machines, install the <code>-cu11</code> wheels, and for CUDA 12.x machines install the <code>-cu12</code> wheels.

Reviewer: What does "CUDA 12.x machine" mean? We need to speak in terms of **driver** or **toolkit**. "Machine" is too ambiguous.

Author: Agreed! 

>**Note**: RAPIDS Docker and Conda CUDA 11.x environments function as expected on a system with a CUDA 12.x driver and they include their own toolkit pieces. <br>pip installations require using the matching wheel to the system's installed CUDA toolkit. For CUDA 11.x toolkits, install the <code>-cu11</code> wheels, and for CUDA 12.x tookits install the <code>-cu12</code> wheels. If your installation has a CUDA 12.x driver but a CUDA 11.x toolkit, use the <code>-cu11</code> wheels.

?

Reviewer: > and they include their own toolkit pieces.

change to
> because they include their own toolkit.

Author: I've made these changes 🎉

<div id="pip"></div>

## **pip**

> Beginning with the release of 23.06: cuDF, dask-cuDF, cuML, cuGraph, cuSpatial, RMM, and RAFT CUDA 11 and CUDA 12 pip packages are available on the NVIDIA Index.

Reviewer: I re-ordered this sentence a bit. The "Beginning with..." made sense when pip packages were new, but not so much now that they're stable.

I googled around a bit and it seems like "NVIDIA Python Package Index" is the name that is being used more broadly, not "NVIDIA Index."
```suggestion
RAPIDS 23.06 pip packages of cuDF, dask-cuDF, cuML, cuGraph, cuSpatial, RMM, and RAFT are available for CUDA 11 and CUDA 12 on the NVIDIA Python Package Index.
```
Also, do we need to list cuCIM? Are we missing anything else?

Author: cuCIM is on pypi proper, this should be correct list for what's on pypi.nvidia.com unless we want to include ptxcompiler or something. I wanted to only include items on the selector though. Which makes me realize I need to add a button to the selector for `dask-cudf`.

1. Install WSL2 and the Ubuntu 22.04 package [using Microsoft's instructions](https://docs.microsoft.com/en-us/windows/wsl/install){: target="_blank"}.
2. Install the [latest NVIDIA Drivers](https://www.nvidia.com/download/index.aspx){: target="_blank"} on the Windows host.
3. Log in to the WSL2 Linux instance.
4. Follow [this helpful developer guide](https://docs.nvidia.com/cuda/wsl-user-guide/index.html#cuda-support-for-wsl2){: target="_blank"} and then install the WSL-specific [CUDA 11.8](https://developer.nvidia.com/cuda-11-8-0-download-archive?target_os=Linux&target_arch=x86_64&Distribution=WSL-Ubuntu&target_version=2.0&target_type=deb_local){: target="_blank"} or [CUDA 12.0](https://developer.nvidia.com/cuda-12-0-1-download-archive?target_os=Linux&target_arch=x86_64&Distribution=WSL-Ubuntu&target_version=2.0&target_type=deb_local){: target="_blank"} Toolkit without drivers into the WSL2 instance.

> 	- The CUDA version you install must match the pip CUDA version you select for pip (`-cu11` or `-cu12`)

Reviewer: ```suggestion
	- The installed CUDA Toolkit version must match the pip wheel version (`-cu11` or `-cu12`)
```
